Codec/Archive/Tar/Types.hs

```diff
@@ -321,9 +321,9 @@ fromTarPathToPosixPath :: TarPath -> FilePath
 fromTarPathToPosixPath (TarPath namebs prefixbs) = adjustDirectory $ BS.Char8.unpack $
    if BS.null prefixbs
    then namebs
-   else prefixbs <> BS.Char8.pack [FilePath.Posix.pathSeparator] <> namebs
+   else prefixbs <> BS.Char8.singleton FilePath.Posix.pathSeparator <> namebs
   where
-    adjustDirectory | BS.Char8.pack [FilePath.Posix.pathSeparator] `BS.isPrefixOf` namebs
+    adjustDirectory | BS.Char8.singleton FilePath.Posix.pathSeparator `BS.Char8.isSuffixOf` namebs
```

Comment:
Actually, it does not look to me that you need to adjust directory at all. The previous implementation used to lose the last slash when splitting directories, so it was important to restore it back afterwards. But now you do not trim the last slash from `namebs`, so no reason to add it back. Currently you'll end up with two slashes.